### ISSUE TITLE
If alwaysSyncPlatformPricesOver = true, - product.special_price always 0

### ISSUE THREAD

**Issue by Devors** [DivanteLtd/vue-storefront#2940]
## Current behavior
reset `product.special_price` in both cases

## Expected behavior
do not reset if `product.priceInclTax < product.originalPriceInclTax`

## Repository
https://github.com/DivanteLtd/vue-storefront/blob/v1.9.0/core/modules/catalog/helpers/index.ts#L158

## Can you handle fixing this bug by yourself?
By removing `else` case?
- [X] YES
- [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [X] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

**Comment by pkarw** [comment]
I belive this is the price that cames from `render-list` Magneto2 endpoint. Any ideas how to solve it?

Debugging starting point: `taxcalc.ts`

**Comment by janmyszkier** [comment]
:thinking: the code that @Devors seems to be a mess. Let's break this
```
if (product.priceInclTax >= product.originalPriceInclTax) {
    product.specialPriceInclTax = 0
    product.special_price = 0
  } else {
    product.special_price = 0 // the same price as original; it's not a promotion
  }
```
into few points....
1. the first bad thing is that the comment in `else` is misleading, says `same price as original`, but the IF was `=` so else clearly takes care of NOT SAME case

2. Considering `product.priceInclTax` is in fact `final_price` (https://github.com/DivanteLtd/vue-storefront/blob/v1.9.0/core/modules/catalog/helpers/index.ts#L142) 
and `originalPriceInclTax` is in fact `regular_price` (https://github.com/DivanteLtd/vue-storefront/blob/v1.9.0/core/modules/catalog/helpers/index.ts#L143) I'd love to know how on earth we woul ever get a promotion that makes the `final_price` (which is AFTER applying promotions) higher than `regular_price`

3. If we are only interested in camelcased values (so `final_price` equivalent) and it was decided to reset special_price in every case (in BOTH `if` and `else`) then why we are checking it in [Product.vue](https://github.com/DivanteLtd/vue-storefront/blob/master/src/themes/default/pages/Product.vue#L35)
? 

**My guess**:
Since VueSF does not support anything except coupons (which is supported only in cart endpoint anyway and is not reflected on product/category pages) we should only simplify this whole if/else into
```
if (product.priceInclTax < product.specialPriceInclTax) {
    product.specialPriceInclTax = 0
    product.special_price = 0
  } 
```
because it's the only case when the price was SOMEHOW lowered by some rule, but lowering the price is not caused by the `special_price` setting by magento. THIS should not show the product was discounted. this way Product.vue would still be correct when looking at special_price.

**Comment by pkarw** [comment]
@janmyszkier, @Devors this condition seems like a logic error. We probably should simplify to:

```js
if (product.priceInclTax >= product.originalPriceInclTax) {
    product.specialPriceInclTax = 0
    product.special_price = 0
  }
```

priceInclTax = final_price

This condition means like: if the final price isn't lower than the original price - there is no `special_price`

Please test it and if it fixes the issue please provide us with a PR

**Comment by pkarw** [comment]
Fixed in the #3043
